src/decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.first(10).join(' ')
```

Comment:
- [ ] After fixing the errors, you'd realize your trimmer does not also work as expected. Kindly fix that.

  You can resolve this by refactoring to this:
  ```
  @nameable.correct_name[0...10]
  ```

  ![tti](https://user-images.githubusercontent.com/36542986/195353984-17accab7-73e8-4864-aa7c-85269dc83187.jpg)